### ISSUE TITLE
Unable to Enumerate User Followed by Python Error

### ISSUE THREAD

**Issue by KyanHexagon** [sse-secure-systems/TeamsEnum#3]
Hello,

I was able to use the tool to successfully enumerate users but after working without issue for some time I started to get the following error for almost every username tried:

```Exception in thread Thread-15 (enumerate_user):
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/share/TeamsEnum/TeamsEnum.py", line 27, in enumerate_user
    enum.check_user(email.strip(), accounttype, presence=presence, outfile=outfile)
  File "/share/TeamsEnum/teamsenum/enum.py", line 42, in check_user
    self.check_teams_user(email, presence, outfile)
  File "/share/TeamsEnum/teamsenum/enum.py", line 88, in check_teams_user
    if presence and "mri" in user_profile[0]:
                             ~~~~~~~~~~~~^^^
KeyError: 0

**Comment by xopham** [comment]
@KyanHexagon we believe that this was caused by malformed email addresses which cause the API to return an unexpected response. @bka-dev just created and merged #4  which handles non-200 status codes better and tests the data structure of the response.
Could you let us no whether the latest version fixes your poblem?

**Comment by KyanHexagon** [comment]
Apologies for the late reply. I haven't experienced this issue again.

I appreciate your help.
